openqaoa/optimizers/result.py

```diff
@@ -108,6 +110,48 @@ def __init__(
 
     #     return (string)
 
+    def asdict(self, keep_cost_hamiltonian:bool=True, complex_to_string:bool=False):
+        """
+        Returns a dictionary with the results of the optimization, where the dictionary is serializable.
+
+        Parameters
+        ----------
+        keep_cost_hamiltonian: `bool`
+            If True, the cost hamiltonian is kept in the dictionary. If False, it is removed.
+        complex_to_string: `bool`
+            If True, the complex numbers are converted to strings. If False, they are kept as complex numbers. This is useful for the JSON serialization.
+
+        Returns
+        -------
+        return_dict: `dict`
+            A dictionary with the results of the optimization, where the dictionary is serializable.
+        """
+
+        return_dict = {}
+        return_dict['method'] = self.method
+        if keep_cost_hamiltonian: return_dict['cost_hamiltonian'] = convert2serialize(self.cost_hamiltonian)
+        return_dict['evals'] = self.evals
+        return_dict['most_probable_states'] = self.most_probable_states
+
+        complx_to_str = lambda x: str(x) if isinstance(x, np.complex128) or isinstance(x, complex) else x
+        
+        if complex_to_string and issubclass(self.__backend, QAOABaseBackendStatevector):
```

Comment:
The docstring should explain this conditional statement, as it is rather not trivial!

Reply:
done